src/App.tsx

```diff
@@ -2,49 +2,75 @@ import React, { useState } from "react";
 import keyboardMap from "./lib/keyboard-map";
 
 function App() {
-  const [keyboardKey, setKeyboardKey] = useState("");
-
-  document.addEventListener("keydown", (e) => {
-    setKeyboardKey(e.key);
-  });
-
-  const renderKeyDictionary = () => {
-    if (!keyboardMap[keyboardKey]) {
-      return (
-        <span className="flex items-center">
-          <strong className="text-xl mr-2 border border-emerald-500 py-1 px-3 rounded">
-            {keyboardKey}
-          </strong>
-          does not have an associated vim command
-        </span>
-      );
-    }
-
-    return (
-      <span className="flex items-center">
-        <strong className="text-xl mr-2 border border-emerald-500 py-1 px-3 rounded">
-          {keyboardKey}
-        </strong>
-        {keyboardMap[keyboardKey]}
-      </span>
-    );
-  };
-
-  return (
-    <div className="App h-full text-light-50 flex items-center justify-center">
-      {keyboardKey.length > 0 ? (
-        <div>{renderKeyDictionary()}</div>
-      ) : (
-        <div>
-          <h1 className="text-5xl mb-4">
-            <span className="text-emerald-500">Vim</span>
-            <span>mer</span>
-          </h1>
-          <p>Press a keyboard key to find what it maps to in Vim.</p>
-        </div>
-      )}
-    </div>
-  );
+	const [keyboardKey, setKeyboardKey] = useState("");
+
+	document.addEventListener("keydown", (e) => {
+		const irreleventKeys = ["Shift", "Control", "Meta"];
+		if (!irreleventKeys.includes(e.key)) {
+			setKeyboardKey(e.key);
+		}
+	});
+
+	const isLetter = (c: string) => {
+		return c.toLowerCase() !== c.toUpperCase();
+	};
+
+	const renderKeyDictionary = () => {
+		const shiftIncluded =
+			keyboardKey === keyboardKey.toUpperCase() && isLetter(keyboardKey) ? (
```

Comment:
This logic determines if you should add `Shift` + `Letter` for uppercase letters. The reason I added the `isLetter` function is that `keyboardKey === keyboardKey.toUpperCase()` is true for non-alphabet letters, so it would output `Shift + ;` if we just typed `;`

Reply:
This feels a little odd to me and I'm not sure if I follow.

Reply:
It is, but I'm not exactly sure what the best approach would be. I can explain better when next time we call